mxcubecore/HardwareObjects/Microdiff.py

```diff
@@ -406,22 +407,46 @@ def _wait_ready(self, timeout=None):
                 time.sleep(0.5)
 
     def open_detector_cover(self, timeout=10):
-        try:
-            detcover = self.get_object_by_role("controller").detcover
+        use = True
+        if self.handle_detector_cover:
+            try:
+                use = self.handle_detector_cover.get_value().value
+            except AttributeError:
+                use = False
 
-            if detcover.state == "IN":
-                detcover.set_out(timeout)
-        except AttributeError:
-            logging.getLogger("HWR").exception("No detector cover configured")
+        logging.getLogger("user_level_log").info("Use detecot cover open: %s" % use)
```

Comment:
`detecot`? I guess it is supposed to be `detector`, right?

```suggestion
        logging.getLogger("user_level_log").info("Use detector cover open: %s" % use)
```